packages/@react-spectrum/form/stories/Form.stories.tsx

```diff
@@ -686,3 +708,40 @@ function FormWithSubmit() {
     </Form>
   );
 }
+
+export const NativeValidation = () => render({
+  isRequired: true,
+  validationBehavior: 'native',
+  showSubmit: true,
+  onSubmit: (e) => {
+    e.preventDefault();
+    action('onSubmit')(Object.fromEntries(new FormData(e.target as HTMLFormElement).entries()));
+  }
+});
+
+NativeValidation.story = {
+  parameters: {description: {data: 'This story is to test that client validation occurs on form submit and updates when the user commits changes to a field value (e.g. on blur).'}}
+};
+
+export function ServerValidation() {
+  let [serverErrors, setServerErrors] = useState<any>({});
+  let onSubmit = async (e) => {
+    e.preventDefault();
+    let errors = {};
+    for (let el of e.target.elements) {
+      errors[el.name] = `Invalid value for "${el.name}".`;
+    }
+    setServerErrors(errors);
+  };
+
+  return render({
+    validationBehavior: 'native',
+    onSubmit,
+    validationErrors: serverErrors,
+    showSubmit: true
+  });
+}
+
+ServerValidation.story = {
+  parameters: {description: {data: 'This story is to test that server errors appear after submission, and are cleared when a field is modified.'}}
```

Comment:
Noticed that the "Favorite Pet" radiogroup validation error after submission doesn't clear when the user modifies the selected radio option via keyboard arrows. It only gets cleared if the user clicks on a different radio option 

To reproduce:
1. go to https://reactspectrum.blob.core.windows.net/reactspectrum/36a3c4cbf1421edd7cb21466d4fdad61077d1ca4/storybook/index.html?path=/story/form--server-validation&providerSwitcher-express=false and hit submit
2. Tab navigate to the "Favorite Pet" radio group and change the selected radio via the keyboard arrows. Note the error doesn't clear